### ISSUE TITLE
Keyword terminators need preceding whitespace, so `SELECT *FROM t` is unparsable

### ISSUE THREAD

**Issue by manan28025** [sqlfluff/sqlfluff#8501]
### Search before asking

- [x] I searched the issues and found no similar issues.

### What Happened

`SELECT *FROM table_a` (no space between `*` and `FROM`) is reported as unparsable. It fails in ansi, postgres, mysql, snowflake, sparksql, databricks and hive. `SELECT t.*FROM table_a t` also fails. bigquery parses it.

This was first reported downstream in reata/sqllineage#776, where the maintainer confirmed Spark accepts this syntax and pointed to sqlfluff's grammar.

### Expected Behaviour

`SELECT *FROM table_a` parses the same way as `SELECT * FROM table_a`.

### Observed Behaviour

```
Line 1, Position 9: Found unparsable section: 'FROM table_a'
```

The cause looks like `greedy_match` in `src/sqlfluff/core/parser/match_algorithms.py`. It only accepts an alphabetical terminator such as `FROM` when whitespace precedes it. Here the previous segment is `*`, so the `FROM` terminator is skipped and the select clause claims the rest of the statement. The comment there already has a TODO to review this rule. The Rust parser applies the same check (`is_preceded_by_whitespace` in `sqlfluffrs/sqlfluffrs_parser/src/parser/table_driven/match_algorithms.rs`), so a fix would likely need to be mirrored there.

Would you accept allowing a keyword terminator when the preceding code segment is a symbol such as `*`, while keeping the whitespace requirement otherwise? I'm happy to open a PR once the intended behaviour is agreed.

### How to reproduce

```python
import sqlfluff

sqlfluff.parse("SELECT * FROM table_a", dialect="ansi")  # parses
sqlfluff.parse("SELECT *FROM table_a", dialect="ansi")   # APIParsingError
```

### Dialect

ansi (also postgres, mysql, snowflake, sparksql, databricks and hive)

### Version

4.3.0. The whitespace rule in `greedy_match` is unchanged on `main`.

### Configuration

None, default configuration.

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

**Comment by peterbud** [comment]
Hi @manan28025 thanks for opening this issue. While execution engines like Spark SQL are lenient and will sometimes parse or execute non-standard syntax due to permissive grammar rules (or fallback mechanisms), SQLFluff's design philosophy is to enforce standardized SQL grammar rather than replicate every engine-specific quirk or parser flaw. If a construct violates standard parsing rules, it's expected to fail linting or parsing until explicitly supported by the dialect grammar.
